Content.Server/Ninja/Systems/NinjaSuitSystem.cs

```diff
@@ -0,0 +1,164 @@
+using Content.Server.Actions;
+using Content.Server.Emp;
+using Content.Server.Popups;
+using Content.Server.Power.Components;
+using Content.Server.PowerCell;
+using Content.Shared.Actions;
+using Content.Shared.Examine;
+using Content.Shared.Hands.EntitySystems;
+using Content.Shared.Ninja.Components;
+using Content.Shared.Ninja.Systems;
+using Content.Shared.Popups;
+using Robust.Shared.Containers;
+
+namespace Content.Server.Ninja.Systems;
+
+public sealed class NinjaSuitSystem : SharedNinjaSuitSystem
+{
+    [Dependency] private readonly ActionsSystem _actions = default!;
+    [Dependency] private readonly EmpSystem _emp = default!;
+    [Dependency] private readonly SharedHandsSystem _hands = default!;
+    [Dependency] private readonly new NinjaSystem _ninja = default!;
+    [Dependency] private readonly PopupSystem _popups = default!;
+    [Dependency] private readonly PowerCellSystem _powerCell = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        // TODO: maybe have suit activation stuff
+        SubscribeLocalEvent<NinjaSuitComponent, ContainerIsInsertingAttemptEvent>(OnSuitInsertAttempt);
+        SubscribeLocalEvent<NinjaSuitComponent, ExaminedEvent>(OnExamined);
+        SubscribeLocalEvent<NinjaSuitComponent, TogglePhaseCloakEvent>(OnTogglePhaseCloak);
+        SubscribeLocalEvent<NinjaSuitComponent, CreateSoapEvent>(OnCreateSoap);
+        SubscribeLocalEvent<NinjaSuitComponent, RecallKatanaEvent>(OnRecallKatana);
+        SubscribeLocalEvent<NinjaSuitComponent, NinjaEmpEvent>(OnEmp);
+    }
+
+    protected override void NinjaEquippedSuit(EntityUid uid, NinjaSuitComponent comp, EntityUid user, NinjaComponent ninja)
+    {
+        base.NinjaEquippedSuit(uid, comp, user, ninja);
+
+        _ninja.SetSuitPowerAlert(user);
+
+        if (!TryComp<ActionsComponent>(user, out var actions))
+            return;
+
+        // TODO: do actions like with gloves are now (before merge)
+        _actions.AddAction(user, comp.TogglePhaseCloakAction, uid, actions);
+        _actions.AddAction(user, comp.RecallKatanaAction, uid, actions);
+        // TODO: ninja stars instead of soap, when embedding is a thing
+        // The cooldown should also be reduced from 10 to 1 or so
+        _actions.AddAction(user, comp.CreateSoapAction, uid, actions);
+        _actions.AddAction(user, comp.KatanaDashAction, uid, actions);
+        _actions.AddAction(user, comp.EmpAction, uid, actions);
+    }
+
+    // TODO: put in shared so client properly predicts insertion, but it uses powercell so how???
```

Comment:
the powercell component itself is in shared. the data about energy isn't, however